include/xtensor/xpad.hpp

```diff
@@ -233,47 +233,80 @@ namespace xt
         return pad(e, pw, mode, constant_value);
     }
 
+    namespace detail {
+
+      template<class E>
+      auto resize(E & e, size_t n) -> decltype(e.resize(n), 1)
+      {
+        e.resize(n);
+        return n;
+      }
+
+      template<class E>
+      auto resize(E &, size_t n) -> decltype(std::tuple_size<E>::value)
+      {
+        XTENSOR_ASSERT(n == std::tuple_size<E>::value);
+        return n;
+      }
```

Comment:
The previous one requires a resize method being available, and this one requires tuple_size being available. It's a way to resize only if there's a resize method, otherwise we check that the static size is ok.

Reply:
Ah, sorry I missed that during the first review. We have tools in the `xtl` to do that (no resize, but initialize with the right size). It does not handle the particular case of `std::tuple` though. Do you have the case where `E` is a tuple? Or is it just a convenient way to get the static size of an `std::array`?

Reply:
yeah it was for std::array.